File: small_problems/easy_4/ex9.rb
DIGITS = { 0 => '0', 1 => '1', 2 => '2', 3 => '3', 4 => '4', 5 => '5',
           6 => '6', 7 => '7', 8 => '8', 9 => '9' }

def integer_to_string(int)
  str = ''
  int.digits.reverse.each do |digit|
    str << DIGITS[digit]
  end
  str
end

p integer_to_string(4321) == '4321'
p integer_to_string(0) == '0'
p integer_to_string(5000) == '5000'

File: js_fundamentals/basics/ex9.js
const DIGITS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];

function integerToString(number) {
  let negative = false;
  if (number !== Math.abs(number)) {
    number = Math.abs(number);
    negative = true;
  }

  let eachDigit = [];
  while (number > 9) {
    let digit = DIGITS[number % 10];
    eachDigit.push(digit);

    number = Math.floor(number / 10);
  }
  eachDigit.push(DIGITS[number]);

  if (negative) {
    eachDigit.push('-');
  } else if (eachDigit.length > 1 || eachDigit[0] !== '0') {
    eachDigit.push('+');
  }

  return eachDigit.reverse().join('');
}

console.log(integerToString(4321));      // "4321"
console.log(integerToString(-123));         // "0"
console.log(integerToString(0));      // "5000"
